frontend/src/types/common/common.ts

```diff
@@ -7,8 +7,13 @@ export type EngineType = "MYSQL" | "POSTGRES" | "ORACLE" | "SQLSERVER";
 // "" meas empty cloud provider
 export type CloudProvider = "AWS" | "ALIYUN" | "GCP" | "";
 
-export const isValidCloudProvider = (val: string): boolean => {
-  return val === "AWS" || val === "ALIYUN" || val === "GCP" || val === "";
+export const isValidCloudProvider = (providerList: string[]): boolean => {
+  providerList.forEach((provider) => {
+    if (provider !== "AWS" && provider !== "GCP" && val !== "") {
+      return false;
```

Comment:
Shall we use for-loop instead of forEach? Does this "return false" only returns the value for forEach callback instead of for the isValidCloudProvider()?

Reply:
This is huge BUG.😭